lua/lualine/component.lua

```diff
@@ -274,11 +283,17 @@ function M:draw(default_highlight, is_focused)
   if self.options.fmt then
     status = self.options.fmt(status or '')
   end
-  if type(status) == 'string' and #status > 0 then
+  local show_separators_if_empty = self.options.show_separators_if_empty
+  if type(show_separators_if_empty) == 'function' then
```

Comment:
when is it useful to have this as a function? It doesn't seem like something that will change much from the setup call. And if you need to check a condition to set it you can just use that function to get the value during setup. like

```lua
require'lualine'.setup {
  options = {
    show_separators_if_empty = some_condition_check()
   }
}
```

Reply:
You're not wrong. It was just so easy to check for that case that I went for it.
I can remove that if you want; in that case, do you prefer the `'function'` case to be silently ignored or should I `print` an error to the user?

Reply:
ignore it. function is implicitly true.